R/register.R

```diff
@@ -258,12 +260,33 @@ get_call_entries <- function(path) {
   close(con)
 
   start <- grep("/* .Call calls */", res, fixed = TRUE)
+
   end <- grep("};", res, fixed = TRUE)
 
   if (length(start) == 0) {
     return("")
   }
-  res[seq(start, end)]
+
+  funs$package <- package
+
+  redundant <- as.character(glue_collapse_data(funs, 'extern SEXP _{package}_{name}', sep = '|'))
+
+  if (nchar(redundant) > 0) {
+    res <- res[!startsWith(res, redundant)]
```

Comment:
I'm not sure this will work properly if `redundant` had more than one search term, e.g.

```r
startsWith(c("foo", "bar"), "foo|bar")
#> [1] FALSE FALSE
# I think you need to use something like this
grepl("^(foo|bar)", c("foo", "bar"))
[1] TRUE TRUE
```

We should also have at least one test that tests for the case with multiple redundant names.

Reply:
Ugh you're right thank you!